app/Services/ImportServiceProvider.js

```diff
@@ -0,0 +1,103 @@
+const { Group } = require('../../database');
+const { ForeignKeyConstraintError } = require('sequelize');
+const ApiError = require('../utils/ApiError.js');
+const httpStatus = require('http-status');
+const { createDrawers } = require('./DrawerServiceProvider.js');
+const { createLanguagePackage } = require('./LanguagePackageServiceProvider.js');
+const { drawers } = require('../utils/constants.js');
+const { createVocabularyCard, createTranslations } = require('./VocabularyServiceProvider.js');
+
+async function storeGroupVocabulary(
+  { name, description, VocabularyCards },
+  userId,
+  languagePackageId,
+  active,
+  activate
+) {
+  try {
+    const group = await Group.create({
+      userId,
+      languagePackageId,
+      name,
+      description,
+      active,
+    });
+
+    VocabularyCards.forEach(async (vocabularyCard) => {
+      const createdCard = await createVocabularyCard(
+        languagePackageId,
+        group.id,
+        vocabularyCard.name,
+        vocabularyCard.description,
+        userId,
+        active,
+        activate
+      );
+      // parse vocabulary card id from response and create translations
+      createTranslations(vocabularyCard.Translations, userId, languagePackageId, createdCard.id);
+    });
```

Comment:
I used a Promise

Reply:
I think the bulkInsert solution would be more perfomant for users who import thausends of vocabularies, because that solution only uses one query to store all data. The other solution uses for each vocabulary one query.